src/Microsoft.Deployment.DotNet.Releases/src/Product.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Deployment.DotNet.Releases
+{
+    /// <summary>
+    /// Provides an overview of a single product, including information related to its support level and the latest SDK and runtime releases.
+    /// </summary>
+    public class Product
+    {
+        /// <summary>
+        /// The version of the product, e.g "5.0" or "1.1".
+        /// </summary>
+        [JsonProperty(PropertyName = "channel-version")]
+        public string ProductVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The end-of-life (EOL) date for this <see cref="Product"/> when it is considered to be out of support. The value 
+        /// may be <see langword="null" /> if the EOL date is undetermined, e.g. when a product is still a prerelease.
+        /// </summary>
+        [JsonProperty(PropertyName = "eol-date")]
+        public DateTime? EndOfLifeDate
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// <see langword="True"/> if the latest release of the product contained a security update;
+        /// <see langword="false"/> otherwise.
+        /// </summary>
+        [JsonProperty(PropertyName = "security")]
+        public bool LatestReleaseIncludesSecurityUpdate
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The date of the latest release for this product.
+        /// </summary>
+        [JsonProperty(PropertyName = "latest-release-date")]
+        public DateTime? LatestReleaseDate
```

Comment:
Some of the older data, especially around 1.0 and 1.1 is not always consistent or complete. There's been instances where fields were either absent, empty or explicitly set to null or "null".

Reply:
Ugh, maybe that old data should be cleaned up? If these are JSON files sitting around somewhere, we should update them rather than making fragile models that represent them. Consumers are going to want to know these things.

Reply:
1.0/1.1 released before the releases.json existed. Some issues in the manifest have been resolved. We could do a scan to see if any quirks remain. @leecow to comment on what our options are around updating the .json files. The other thing to note is that we know there are external consumers that have likely worked around some of the data that is incorrect, so there is risk in fixing the older entries.

Reply:
Yes, backwards compatibility is always important. This is why I believe relying on the JSON makes this entire approach fragile.

Reply:
@joeloff - I don't have any material objection to fixing up old data, particularly where it may be leading you to implement problematic workarounds.

Reply:
Thanks @leecow I'll remove the nullable datetime and do a pass over the current data.

Reply:
I've fixed this now.